sensors/depth/depth.go

```diff
@@ -0,0 +1,53 @@
+// Package depth implements the Depth sensor
+package depth
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/registry"
+
+	"go.viam.com/slam/config"
+	"go.viam.com/slam/sensors"
+	"go.viam.com/slam/sensors/utils"
+)
+
+// Depth represents a depth sensor.
+type Depth struct {
+	Name         string
+	depth        camera.Camera
+	DataRateMsec int
```

Comment:
Same question. I don't have a good understanding of how `depth|rgb|lidar|.DataRateMsec` & `sensor.DefaultDataRateMs` are intended to interact with one another & why this makes sense: https://github.com/viamrobotics/slam/pull/160/files#diff-255612532e71d064ed5e0725da530352d9e537d7410e071ace9aa15b3f7f8c36R28

Reply:
Will remove those! I designed this with an older version of the movement scope doc in mind.